developer_manual/general/publishing.rst

```diff
@@ -0,0 +1,131 @@
+
+note:
+Official apps include those part of the release tarball. We'd like to keep the tarball minimal so most official apps are not part of the standard installation.
+
+Approved
+--------
+Approved apps are developed by trusted developers and have passed a cursory security check. They are actively maintained in an open code repository and their maintainers deem them to be stable for casual to normal use.
+
+Requirements:
+
+* code is developed in an open and version-managed code repository, ideally github with git but other scm/hosting is OK.
+* minimum of one active developer/maintainer
+* minimum 5 ratings, average score 60/100 or better
+* app is at least 3 months old
+* developer trust:
+ * developer is known in community;
+ * app is subject to unannounced security audits
+ * in practice this means that at least some of the code of this developer has been audited; either through another app by the same developer or with an earlier version of the app. And that the attitude of the developer towards these audits has been positive.
+* follows app guidelines
+* app is signed, at least domain verified
+
+App store:
+
+* visible in app store by default
+* sorted above experimental apps
+* search results sorted by ratings
+* developer can directly push new versions to the store
+* warning shows for security/stability risks
+
+Experimental
+------------
+Apps which have not been checked at all for security and are new or known to be unstable and under heavy development but are considered usable for casual use or experimentation.
+
+Requirements:
+
+* no malicious intent found from this developer at any time
+* 0 confirmed security problems
+* less than 3 unconfirmed 'security flags'
+* rating over 20/100
+* app is signed but no verification has to be done
+
+App store:
+
+* show up in Apps page provided user has enabled "allow installation of experimental apps" in the settings.
+* Warning about security and stability risks show for app
+* sorted below all others.
+
+Getting an app approved
+=======================
+If you want your app to be approved, make sure you fulfill all the requirements and send an email to XXXX. A team of core ownCloud developers will review your application.
+
+Losing a rating
+---------------
+Apps can lose their rating when:
+
+* they are found to no longer satisfy the requirements
+* when security/malicious intent issues are found
+* when developer requests so
+
+App guidelines
+==============
+These are the app guidelines an app has to comply with to have a chance to be approved.
+
+Legal and security
+
+* Apps can not use 'ownCloud' in their name
+* Irregular and unannounced security audits of all apps can and will take place. If ever any indication of malicious intent is found, the developer(s) can count on a minimum 2 year ban from any ownCloud app infrastructure. Malicious intent includes deliberate spying on users by leaking user data to a third party system, ... (lukas, help out, man! what is evil?)
```

Comment:
I'd make sure to ensure here to mention that if someone publishes an application in bad faith that leaks files or adds by intent backdoors to ownCloud this is considered as malicious intent.

Furthermore we always give the author the benefits of not knowing, so if an application has a security bug: Not a problem. This is not bad faith. If they fix it in time: Everything is fine. But if somebody adds a backdoor (i.e. an hard-coded user account) this might be interpreted as bad faith.

Furthermore I'd mention that distributing malicious applications may have legal consequences (not only from us but also from the users!)